src/nodes/Token.hack

```diff
@@ -121,6 +121,7 @@ abstract class Token extends Node {
     $leading_list = __Private\fold_map(
       /* HH_FIXME[4110] use like-types when available*/ $json['leading'],
       ($j, $p) ==> Trivia::fromJSON($j, $file, $p, $source, 'Node'),
+      /* HH_FIXME[4297] use like-types when available*/
```

Comment:
Like-types won't help with non-nested stuff - `as shape('width' => int, ...)` will work here.

Reply:
```
A linter threw an exception:
  Linter: Facebook\HHAST\AsyncFunctionAndMethodLinter
  File: /opt/lampp/htdocs/hhvm_repos/hhast/src/find_type_for_node.hack
  Exception: Facebook\HHAST\ASTDeserializationError
  Message: In file "/opt/lampp/htdocs/hhvm_repos/hhast/src/find_type_for_node.hack": Failed to deserialize AST: Expected shape('width' => int, ...), got HH\dict
```

This does not work unless the runtime setting for shape()'s are dict's is turned on.
The as fails at runtime.

Reply:
```HACK
$j as KeyedContainer<_, _>['width'] as int
```

is almost the same thing as `$j as shape('width' => int, ...)['width']`, but it works with dicts too.